docs/guides/circuit_runners.rst

```diff
@@ -0,0 +1,200 @@
+
+.. literalinclude:: ../examples/circuit_runners_guide.py
+  :start-after: # IBMQ runner creation example
+  :end-before: # End IBMQ runner creation example
+
+For an example of direct construction of `QiskitRunner`, we will just wrap an Aer backend. In a similar fashion, one can construct a ``WavefunctionSimulator``, provided that the passed backend supports computing state vector.
+
+.. literalinclude:: ../examples/circuit_runners_guide.py
+  :start-after: # Example Qiskit options
+  :end-before: # End WavefunctionSimulator creation example
+
+
+General CircuitRunner methods
+-----------------------------
+
+Once the initialization is done, we can send the ``Circuit`` and number of repetitions to the runner and read the measurements. For more information on creating circuits to send to a circuit runner, refer to the :ref:`Circuits Guide <circuits_guide>`. For a single circuit, that looks like this:
+
+.. literalinclude:: ../examples/circuit_runners_guide.py
+  :start-after: # CircuitRunner run and measure circuit
+  :end-before: # End CircuitRunner run and measure circuit
+
+If you want to run a batch of multiple circuits, possibly with different number of samples, the code needs to be modified as follows:
+
+.. literalinclude:: ../examples/circuit_runners_guide.py
+  :start-after: # CircuitRunner run batch and measure
+  :end-before: # End CircuitRunner run batch and measure
+
+``CircuitRunner`` can also provide the distribution of the outcome measurement. It can be extracted using the following approach:
+
+.. literalinclude:: ../examples/circuit_runners_guide.py
+  :start-after: # CircuitRunner measurement distribution
+  :end-before: # End CircuitRunner measurement distribution
+
+Notice that in all instances, ``CircuitRunner`` automatically adds the measurement operators to the circuit. Therefore, the user is not required to include them in the circuit.
+
+WavefunctionSimulator-specific methods
+--------------------------------------
+
+``WavefunctionSimulator`` can be used to extract a wave function and expectation value for a given circuit (or circuit and operator). Below is an example of extracting these values using a ``WavefunctionSimulator`` called :class:`CirqSimulator <orquestra.integrations.cirq.simulator.CirqSimulator>`:
+
+.. literalinclude:: ../examples/circuit_runners_guide.py
+  :start-after: # WavefunctionSimulator examples
+  :end-before: # End WavefunctionSimulator examples
+
+**Noise Models** can be used on certain simulators (like Qiskit and QSim) to help evaluate a circuit's performance under noisy conditions. For more details how to use that, please refer to docs for particular simulator.
```

Comment:
Add sth that we don't have a common interface for noise models across all our integrations, they are integration-specific.